src/main/java/me/qoomon/gradle/gitversioning/GitVersioningPluginExtension.java

```diff
@@ -603,6 +598,17 @@ private Map<String, Supplier<String>> generateGlobalFormatPlaceholderMap(GitSitu
         return placeholderMap;
     }
 
+    private Matcher matchVersion(String input) {
```

Comment:
I think I found the root cause of the matching problem.
```java
private static final Pattern VERSION_PATTERN = Pattern.compile("(:?(?<core>(?<major>\\d+)(:?\\.(?<minor>\\d+)(:?\\.(?<patch>\\d+))?)?)(:?-(?<label>.*))?)?");
// should be just
private static final Pattern VERSION_PATTERN = Pattern.compile("(?<core>(?<major>\\d+)(:?\\.(?<minor>\\d+)(:?\\.(?<patch>\\d+))?)?)(:?-(?<label>.*))?");
```

`matchVersion` function should look like this then
```java
private static Matcher matchVersion(String input) {
        Matcher matcher = VERSION_PATTERN.matcher(input);
        //noinspection ResultOfMethodCallIgnored
        if(matcher.find()){
            return matcher;
        }

        Matcher fallbackMatcher = Pattern.compile("(:?" + matcher.pattern() + ")?").matcher(input);
        fallbackMatcher.find();
        return fallbackMatcher;
    }
```

Reply:
Made a change with a few minor tweaks.